dkms.in

```diff
@@ -2216,24 +2214,15 @@ autoinstall() {
             installed_modules[${#installed_modules[@]}]="$m"
             continue
         fi
-        # if the module does not want to be autoinstalled, skip it.
-        read_conf_or_die "$k" "$a" "$dkms_tree/$m/$v/source/dkms.conf"
+        # If the module does not want to be autoinstalled, skip it.
+        read_conf_or_die "$kernelver" "$arch" "$dkms_tree/$m/$v/source/dkms.conf"
         if [[ ! $AUTOINSTALL ]]; then
             continue
         fi
-        # otherwise, only autoinstall the latest version we have hanging around.
-        if [[ ($(VER $v) > $(VER $last_v)) ]]; then
-            last_v="$v"
-            tenative="$m/$v"
-        fi
-    done < <(module_status)
-
-    # We may have exited the loop with $tenative set.  If it is,
-    # it contains something that should be updated.
-    if [[ $tenative ]]; then
-        to_install[${#to_install[@]}]="$tenative"
+        # Otherwise, autoinstall the latest version we have hanging around.
+        to_install[${#to_install[@]}]="$m/$v"
         build_depends["$m"]="${BUILD_DEPENDS[@]}"
-    fi
+    done
```

Comment:
At a glance - this patch seems unrelated to the rest of the PR. Let's split it a separate PR?

Reply:
Right, the `simplify` threw me off the scent for a moment.